diff --git a/include/configs/sun8iw7p1.h b/include/configs/sun8iw7p1.h
index f97709c..139c946 100644
--- a/include/configs/sun8iw7p1.h
+++ b/include/configs/sun8iw7p1.h
@@ -331,7 +331,7 @@
 	"init=${init} loglevel=${loglevel} partitions=${partitions}\0" \
 	"setargs_mmc=setenv bootargs console=${console} root=${mmc_root} " \
 	"init=${init} loglevel=${loglevel} partitions=${partitions}\0" \
-	"boot_normal=fatload mmc 0:1 0x43000000 script.bin;fatload mmc 0:1 40007800 uimage;bootm 40007800\0" \
+	"boot_normal=fatload mmc 2:1 0x43000000 script.bin;fatload mmc 2:1 40007800 uimage;bootm 40007800\0" \
 	"boot_recovery=sunxi_flash read 40007800 recovery;boota 40007800\0" \
 	"boot_fastboot=fastboot\0"
 
